HIS/his_checker.py

```diff
@@ -0,0 +1,177 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (c) Bao Project and Contributors. All rights reserved
+
+import os
+import sys
+import subprocess
+
+cyclomatic_treshold = 10
+comment_density = 0.2
+goto_treshold = 0
+call_treshold = 7
+callee_treshold = 5
+statements_treshold = 50
+
+def process_metrics(filename):
+
+    """Process each HIS metric."""
+
+    process_complexity(filename)
+    process_comments(filename)
+    process_statements(filename)
+    process_callees(filename)
+    process_goto(filename)
+    process_calls(filename)
+
+def process_complexity(file):
+
+    """Process McCabe Cyclomatic Complexity for each function in a file"""
+
+    complexity = "pmccabe -c "
+    lines = os.popen(complexity + str(file))
+    lines = lines.read()
+    sline = lines.split('\n')
+
+    for num in range(len(sline)-1):
+        fields = sline[num].split('\t')
+        colonyCount = int(fields[0])
+        if colonyCount > cyclomatic_treshold:
+            eprint("Complexity exceeded (max." + str(cyclomatic_treshold) + "): "
```

Comment:
eprint is not a standard function. To use it, you must import the eprint library (from eprint import eprint) or use the sys.stderr file.

Reply:
Ended up replacing it with _print_